src/ansys/mapdl/core/mapdl_extended.py

```diff
@@ -1214,10 +1267,28 @@ def eplot(self, show_node_numbering=False, vtk=None, **kwargs):
                 )
 
         if vtk:
+            import pyvista as pv
+
+            from ansys.mapdl.core.plotting.visualizer import MapdlPlotter
+
+            if "plotter" not in kwargs:
+                pl = MapdlPlotter()
+            else:
+                pl = kwargs.pop("plotter")
+                if not (isinstance(pl, MapdlPlotter) or isinstance(pl, pv.Plotter)):
+                    raise TypeError(
+                        f"Expected a MapdlPlotter instance, but got {type(pl)}"
+                    )
+                elif isinstance(pl, pv.Plotter):
+                    mapdl_pl = MapdlPlotter()
+                    mapdl_pl.switch_scene(pl)
+                    pl = mapdl_pl
+
```

Comment:
I keep seeing this piece of code everywhere (this is like the fourth time).

May I suggest?
```suggestion
            import pyvista as pv

            from ansys.mapdl.core.plotting.visualizer import MapdlPlotter

            pl = kwargs.get("plotter", None)
            pl = MapdlPlotter().switch_scene(pl)

```

This would require that:
- the visualizer checks the object that is passed to ``switch_scene``. I think that is a task for the visualizer, check what kind of object it can receive.
- ``switch_scene`` should accept ``None`` in which case, it returns an empty plotter. Alternatively, you could have otherwise:
  ```py
            pl = kwargs.get("plotter", None)
            if pl:
                pl = MapdlPlotter().switch_scene(pl)
            else:
                pl = MapdlPlotter()
  ```

I leave it up to you. This is merely a suggestion.